src/main/java/jalasoft/com/models/ToDoDictionary.java

```diff
@@ -0,0 +1,38 @@
+package jalasoft.com.models;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import jalasoft.com.utils.Constants;
+import jalasoft.com.utils.DictionaryUtils;
+
+public class ToDoDictionary extends Dictionary<String> {
+
+	DictionaryUtils dictionaryUtils;
+	Set<String> words;
+
+	public ToDoDictionary() {
+		dictionaryUtils = new DictionaryUtils();
+		setItems();
+	}
+
+	@Override
+	public String search(String wordToSearch) {
+		String resultWords = Constants.EMPTY;
+		for (String word : items) {
+			if (word.equals(wordToSearch)) {
+				resultWords = wordToSearch;
+			}
+		}
+		return resultWords;
+	}
+
+	@Override
+	public void setItems() {
+		items = new HashSet<String>();

```

Comment:
if you add a new TODO task, how the dictionary is handled?

Reply:
The dictionary takes data from the ToDoListPersister that is a singleton. So, in case a new task is added or deleted we can instantiate a new dictionary or just call setItems() again.

Reply:
where is that logic?